src/plugins/social-sharing-mixin.js

```diff
@@ -0,0 +1,136 @@
+import { getFullUrl } from '@/util/urlUtils';
+import clipboardCopy from 'clipboard-copy';
+
+/* This mixin depends on the component using it having the
+* following data or computed variables available:
+* this.shareMessage and this.shareLink
```

Comment:
I tried to make this as generic as possible for this first pass, but it still depends on shareMessage and shareLink from the component